cpp/src/parquet/encoding.cc

```diff
@@ -3503,7 +3591,11 @@ std::unique_ptr<Decoder> MakeDictDecoder(Type::type type_num,
     case Type::DOUBLE:
       return std::make_unique<DictDecoderImpl<DoubleType>>(descr, pool);
     case Type::BYTE_ARRAY:
-      return std::make_unique<DictByteArrayDecoderImpl>(descr, pool);
+      if (use_large_binary_variants) {
+        return std::make_unique<DictLargeByteArrayDecoderImpl>(descr, pool);
+      } else {
+        return std::make_unique<DictByteArrayDecoderImpl<>>(descr, pool);
```

Comment:
aghh just a thing of the moment because the generic class is already named like a non large variant, so I couldn't come up with an alias. If you have an alias in mind, let me know and I'll create it.

Reply:
Ok I guess maybe just `DictByteArrayDecoderImpl<ByteArrayType>` is ok?